modules/cnn_3dobj/src/cnn_feature.cpp

```diff
@@ -0,0 +1,244 @@
+#include "precomp.hpp"
+using namespace caffe;
+using std::string;
+
+namespace cv
+{
+namespace cnn_3dobj
+{
+    descriptorExtractor::descriptorExtractor(const string& device_type, int device_id)
```

Comment:
the device_id should be optional (bacause in the CPU case, we don't need it), with a default of 0.